homeassistant/components/aladdin_connect/sensor.py

```diff
@@ -89,14 +98,17 @@ def __init__(
         device: DoorDevice,
         description: AccSensorEntityDescription,
     ) -> None:
-        """Initialize a sensor for an Abode device."""
+        """Initialize a sensor for an Aladdin Connect device."""
         self._device_id = device["device_id"]
         self._number = device["door_number"]
         self._name = device["name"]
+        self._model = device["model"]
         self._acc = acc
         self.entity_description = description
         self._attr_unique_id = f"{self._device_id}-{self._number}-{description.key}"
         self._attr_has_entity_name = True
+        if self._model == "01" and description.key in ("battery_level", "ble_strength"):
+            self._attr_entity_registry_enabled_default = True
```

Comment:
The `description` being passed in has a `entity_registry_enabled_default` property that's essentially being ignored here.  Could we change the logic to remove the `description.key` check and just set 

```python
self._attr_entity_registry_enabled_default = device.entity_registry_enabled_default
```

And then change the two sensors which _should_ be enabled by default to true in the definitions above?  Otherwise, what's the point of the property in the description?

Reply:
99% of the devices don't use these two sensors (BLE & battery)- the exception is model 01 to enable.  
I may mis-understand, but the key is used when the user overrides and enables the device (right?) So these values need to be present.
I preferred to have WIFI_RSSI enabled by default, but other HA members say it's "not needed" and more of a nice to have - so it's disabled by default.

Reply:
Maybe instead of using registry-enabled we should instead change the entity-category to diagnostics.
Then all users can simply enable them manually inside Home Assistant... without the need for this condition.

Reply:
Simply removing this condition will require the user to enable them
manually inside HA (disabled be default) right?
Maybe diagnostic is the right type?

I was trying to help the user with model 01 know they are useful for their
version, as the values are 0 for other models (aka not obviously useless)

Another idea may be to only have them available when model 01 it's
detected, not available at all for other models. This would require adding
future models in the code of users discover another model that uses these.
Currently they could just manually enabled them. (I'm not sure how to
actually "hide" sensors after the model is detected)

Reply:
I'm fine with either option based on my understanding of the sensors now.